### PR TITLE
[RW-2989][risk=no] Fix group count updates on temporal field change

### PR DESCRIPTION
**Pull request by dolbeew**
- Update group counts when temporal select or input changes
- Fix display and timing of loading spinners and error icons on cohort overview page
- Prevent duplicate api calls when updating items or editing/cloning an existing cohort

### CODE REVIEW

File: ui/src/app/cohort-search/list-overview/list-overview.component.ts [hunk 1]
```diff
@@ -57,7 +57,10 @@ export class ListOverviewComponent implements OnChanges, OnInit {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes.update && this.hasActiveItems && !this.hasTemporalError) {
+    if (changes.update &&
```
Comment (freemabd) [inline ui/src/app/cohort-search/list-overview/list-overview.component.ts]: Could this logic be in a a function that has a more descriptive name so it's easier to understand the condition?

File: ui/src/app/cohort-search/list-overview/list-overview.component.ts [hunk 2]
```diff
@@ -66,13 +69,17 @@ export class ListOverviewComponent implements OnChanges, OnInit {
 
   getTotalCount() {
     try {
+      this.apiCallCheck++;
+      const apiCallCheck = this.apiCallCheck;
```
Comment (freemabd) [inline ui/src/app/cohort-search/list-overview/list-overview.component.ts]: nit: maybe give them different names?

File: ui/src/app/cohort-search/list-overview/list-overview.component.ts (hunk 1 above)
Comment (dolbeew) [reply ui/src/app/cohort-search/list-overview/list-overview.component.ts]: Good idea, done.

File: ui/src/app/cohort-search/list-overview/list-overview.component.ts (hunk 2 above)
Comment (dolbeew) [reply ui/src/app/cohort-search/list-overview/list-overview.component.ts]: Done

**Pull request merged**